File: 2022/Day 1/day1.go
//Go version of Day 1 solution
//slower than python

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	file, err := os.Open("data/day1.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanLines)
	var txtlines []string
	for scanner.Scan() {
		txtlines = append(txtlines, scanner.Text())
	}

	var intlines []int
	for _, eachline := range txtlines {
		i, _ := strconv.Atoi(eachline)
		intlines = append(intlines, i)
	}
	//dont mind the variable names lol
	var getsum int
	var eatsum int
	var groupsum int
	for _, eachline := range intlines {
		if eachline == 0 {
			groupsum++
			if getsum > eatsum {
				eatsum = getsum
			}
			getsum = 0
		} else {
			getsum += eachline
		}
	}
	fmt.Println("Part 1 is", eatsum)
}

File: 2022/Day 1/day1.py
with open("2022\Day 1\data\day1.txt") as f:
    a = []
    b = 0
    for i, line in enumerate(f):
        if line.strip():
            b += int(line.strip())
        else:
            a.append(b)
            b = 0
    p1 = sorted(a,reverse=True)[0]
    p2 = sum(sorted(a,reverse=True)[:3])
    print("Part 1: " + str(p1))
    print("Part 2: " + str(p2))